setup-env/action.yaml

```diff
@@ -20,14 +20,48 @@ runs:
           echo "package_manager=npm" >> $GITHUB_OUTPUT
           echo "install_cmd=npm ci" >> $GITHUB_OUTPUT
           echo "hash_glob=**/package-lock.json" >> $GITHUB_OUTPUT
+          FOUND_NODE=true
         elif [ -e yarn.lock ]; then
           echo "package_manager=yarn" >> $GITHUB_OUTPUT
           echo "install_cmd=yarn install --immutable" >> $GITHUB_OUTPUT
           echo "hash_glob=**/yarn.lock" >> $GITHUB_OUTPUT
+          FOUND_NODE=true
         elif [ -e pnpm-lock.yaml ]; then
           echo "package_manager=pnpm" >> $GITHUB_OUTPUT
           echo "install_cmd=pnpm install --frozen-lockfile" >> $GITHUB_OUTPUT
           echo "hash_glob=**/pnpm-lock.yaml" >> $GITHUB_OUTPUT
+          FOUND_NODE=true
+        fi
+
+        if [ -v $FOUND_NODE ]; then
+          exit 0
+        fi
+
+        if [ -e .nvmrc ]; then
+          echo "node_version_file=.nvmrc" >> $GITHUB_OUTPUT
+          FOUND_NODE_VERSION=1
+        elif [ -e .node-version ]; then
+          echo "node_version_file=.node-version" >> $GITHUB_OUTPUT
+          FOUND_NODE_VERSION=1
+        elif [ -e .tool-versions ]; then
+          echo "node_version_file=.tool-versions" >> $GITHUB_OUTPUT
+          FOUND_NODE_VERSION=1
+        elif [ -e package.json ]; then
+          echo "node_version_file=package.json" >> $GITHUB_OUTPUT
+          FOUND_NODE_VERSION=1
+        fi
+
+        if [ -v $FOUND_NODE_VERSION ]; then
+          exit 0
+        fi
+
+        echo "Found node lockfiles but no node version!"
```

Comment:
maybe I'm misreading this - but won't this script have exited by now if this message was true?

Reply:
I think this is fine to leave in, since this is very valuable debug info for reaching this point. Maybe lower the "volume" though to info (no `!`).

Reply:
Maybe "no node version to install"